services/galley/src/Galley/API/Message.hs

```diff
@@ -492,17 +492,12 @@ postQualifiedOtrMessage senderType sender mconn lcnv msg =
           predicate (d, (u, _)) = any (\(d', (u', _)) -> d == d' && u == u') failed'
           -- Failed users/clients aren't redundant
           (failed, redundant) = partition predicate redundant'
+          collectedFailedToSend = collectFailedToSend [qualifiedUserClients failedToSend, toDomMap unconfirmedUnknownClients, fromDomUserClient failed]
```

Comment:
`collectFailedToSend` would not be needed if we had a proper semigroup instance for `Map` where `a` has a monoid instance. But let's worry about this later.